snuba/clickhouse/translators/snuba/mappers.py

```diff
@@ -0,0 +1,79 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from snuba.clickhouse.translators.snuba import SnubaClickhouseStrictTranslator
+from snuba.clickhouse.translators.snuba.allowed import (
+    ColumnMapper,
+    SubscriptableReferenceMapper,
+)
+from snuba.query.dsl import array_element
+from snuba.query.expressions import Column, FunctionCall, SubscriptableReference
+
+
+@dataclass(frozen=True)
+class SimpleColumnMapper(ColumnMapper):
```

Comment:
These two rules (column and tags) are just example to show how this will work.
See tests/clickhouse/translator/test_translation.py for examples of how to connect everything.

If/when we will introduce more complex expressions in the snuba language (that will require a more complex translation) we could introduce a matcher based ExpressionMapper, that is configured with a matcher that matches complex expression and finds parameters (like a regex) and provides the parameters to an executor that builds the result expression